embassy-time/src/queue_generic.rs

```diff
@@ -96,35 +95,35 @@ impl InnerQueue {
         loop {
             let now = Instant::now();
 
-            while self.queue.peek().filter(|timer| timer.at <= now).is_some() {
-                self.queue.pop().unwrap().waker.wake();
+            let mut next_alarm = Instant::MAX;
+
+            let mut i = 0;
+            while i < self.queue.len() {
```

Comment:
This O(n^2) though.

Reply:
is it? `swap_remove` is `O(1)`, it swaps with the last element instead of shifting all after.

Reply:
Yep - just checked - sorry!